### PR TITLE
Add rubinenv recipe

### PR DESCRIPTION
**Pull request by brianv0**
This is actually a metapackage which defines the packages required to build and run the Rubin observatory's software.

We had been previously giving out users environment files, at first `conda env export` and then `conda list --explicit`, to install, but we have ran into issues around the conda solver being extremely slow when operating on older environment output ("stale" conda solutions) when users want to modify their environments, as well as issues around the conda-meta/history file in how it is factored into the solver. In testing, a metapackage, sometimes with truncation of the history file, was the best way to provide some flexibility to users to install software with some flexibility in what is installed. (We are still iterating on this issue)

Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by timj** [comment]
I am willing to be a maintainer.

**Comment by gcomoretto** [comment]
I am willing to be a maintainer

**Comment by brianv0** [comment]
@conda-forge/help-python-c can I get a look at this?

File: recipes/rubinenv/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    # host-like and run-like
+    # These are mostly run-like, but for the same reasons as
+    # above, we need most of these at build time, so they are mostly
+    # host-like
+    - boost =1.70
+    - cfitsio
+    - eigen
+    - galsim
+    - libaprutil
+    - lapack
```
Comment (beckermr) [inline recipes/rubinenv/meta.yaml:61]: You should not be building against this package. It is deprecated and has not been updated in over a year. The current version is 3.8.X and this is at 3.6.X. See the documentation on blas.

Comment (brianv0) [reply recipes/rubinenv/meta.yaml]: Removing it, I think there was a header in there I needed not in the other packages, but it's find to remove for now and I'll investigate.

File: recipes/rubinenv/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    - apr
+    - autograd
+    - astropy
+    - backoff
+    - boto3
+    - bottleneck
+    # conda is required for runtime inspection of installed software
+    - conda
+    - cffi
+    - click <7.1
```
Comment (timj) [inline recipes/rubinenv/meta.yaml:79]: It's trivial to fix the test problem we have with click 7.1

File: recipes/rubinenv/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+# Since all requirements are required "at run time" for the
+# rubin developers, all requirements are fundamentally
+# run requirements. However, we've attempted to organize 
+# requirements as build-like, host-like, or run-like, in 
+# order inform any evolution of this recipe.
+requirements:
+  # conda-forge-pinning dependencies should also be in host
+  # https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml
+  host:
```
Comment (beckermr) [inline recipes/rubinenv/meta.yaml:21]: This host env won't appear in any downstream environments made with this package or when this package is used in other builds. What is the purpose of the host env?

File: recipes/rubinenv/meta.yaml [hunk 4]
```diff
@@ -76,7 +76,7 @@ requirements:
     # conda is required for runtime inspection of installed software
     - conda
     - cffi
-    - click <7.1
+    - click
```
Comment (timj) [inline recipes/rubinenv/meta.yaml:79]: `>= 7.0` would better match the requirement. We know our code does not work with 6.x.

File: recipes/rubinenv/meta.yaml (hunk 3 above)
Comment (brianv0) [reply recipes/rubinenv/meta.yaml]: misunderstanding, fixed now.

File: recipes/rubinenv/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+# Since all requirements are required "at run time" for the
+# rubin developers, all requirements are fundamentally
+# run requirements. However, we've attempted to organize 
+# requirements as build-like, host-like, or run-like, in 
+# order inform any evolution of this recipe.
+requirements:
+  # We need them as host so we know what's installed
+  run:
+    # jq is special - it is needed for the pinit utility in this package
+    - jq
```
Comment (beckermr) [inline recipes/rubinenv/meta.yaml:22]: where is the utility? I don't see it.

File: recipes/rubinenv/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    # build-like dependencies
+    # As part of our scons build process, we do run tests, so
+    # these really are required to perform a build.
+    # Note that "testing" in conda-forge should be limited testing 
+    # sucessful packaging, and not testing code.
```
Comment (ktlim) [inline recipes/rubinenv/meta.yaml:27]:
```suggestion
    # Note that "testing" in conda-forge should be limited to testing 
    # successful packaging, and not testing code.
```

File: recipes/rubinenv/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+  commands:
+    - echo "I'm a metapackage"
+
+
+about:
+  home: https://github.com/conda-forge/rubinenv-feedstock
+  license: BSD-3-Clause
+  license_family: BSD
+  license_file: LICENSE.txt
+  summary: Metapackage to install select the Rubin observatory's software requirements.
```
Comment (ktlim) [inline recipes/rubinenv/meta.yaml:120]:
```suggestion
  summary: Metapackage to install the Rubin Observatory's common software environment. 
```

File: recipes/rubinenv/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+  home: https://github.com/conda-forge/rubinenv-feedstock
+  license: BSD-3-Clause
+  license_family: BSD
+  license_file: LICENSE.txt
+  summary: Metapackage to install select the Rubin observatory's software requirements. 
+
+  description: | 
+    This metapackage exists to give some flexibility to users who want to customize
+    their environment, with DM being able to do some amount of version pinning to
+    guarantee stability
```
Comment (ktlim) [inline recipes/rubinenv/meta.yaml:125]:
```suggestion
    This metapackage exists to define the Rubin Observatory common software
    environment, including version specifications where needed, while allowing
    users flexibility when installing additional packages into the same environment.
```

**Review by ktlim — COMMENTED** [review]
Minor wording suggestions.

File: recipes/rubinenv/meta.yaml (hunk 5 above)
Comment (brianv0) [reply recipes/rubinenv/meta.yaml:21]: fixed

**Review by beckermr — CHANGES_REQUESTED** [review]
Thanks for all of the hard work here! We should add a test of the pin-it code to the recipe. Then I think it will be done!

**Review by beckermr — APPROVED** [review]

**Comment by brianv0** [comment]
I think we're happy with this. I did a recent build on our jenkins servers and it worked fine.

**Pull request merged**